src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DataAccess.scala

```diff
@@ -48,4 +48,36 @@ trait DataAccess
       workflowFailureQuery.schema ++
       workflowMessageQuery.schema ++
       workflowAuditStatusQuery.schema
+
+  def truncateAll: WriteAction[Int] = {
+    // important to keep the right order for referential integrity !
```

Comment:
IMO this is fine as it is. dynamically determining referential integrity sounds like a lot of work with little reward when we already know the order the tables should be deleted in. unless our schema drastically changes it should be pretty easy to add new tables in the right order here. if someone else objects to forcing an order then I'd say alternative 2 is the right approach

also, truncation seems like a good way to do this